src/components/form/FormInput.tsx

```diff
@@ -1,5 +1,94 @@
 import React from 'react';
+import { UseFormRegister } from 'react-hook-form';
 
-export const FormInput = () => {
-  return <div>FormInput</div>;
+export interface IFormInputProps {
+  warning?: boolean;
+  success?: boolean;
+  statusText?: string;
+  icon?: JSX.Element;
```

Comment:
```suggestion
  icon?: IconType;
```

You can import that type from react-icons like so: `{ IconType }` from '...';